pupil_src/shared_modules/marker_tracker_3d/optimization/model_state.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+
+import numpy as np
+
+import file_methods
+from marker_tracker_3d import utils
+
+logger = logging.getLogger(__name__)
+
+
+class ModelState:
+    def __init__(self, save_path):
+        self.save_path = save_path
+
+        self.current_frame_id = 0
+
+        self.frames_id = []
+        self.markers_id = []
+
+        self.all_novel_markers = []
+
+        self.camera_extrinsics_opt = {}  # {frame id: camera extrinsics(list)}
+        self.marker_extrinsics_opt = {}  # {marker id: marker extrinsics(list)}
+        self.marker_points_3d_opt = {}
+
+        self.model = None
+        self.load_marker_tracker_3d_model_from_file()
+
+    def load_marker_tracker_3d_model_from_file(self):
+        self.model = file_methods.Persistent_Dict(
+            os.path.join(self.save_path, "marker_tracker_3d_model")
+        )
+        self.marker_extrinsics_opt = {
+            k: np.array(v)
+            for k, v in self.model.get("marker_extrinsics_opt", {}).items()
+        }
+
+        try:
+            self.markers_id = [list(self.marker_extrinsics_opt.keys())[0]]
```

Comment:
Maybe `self.markers_id = [next(iter(self.marker_extrinsics_opt.keys())]`? Imo, this has 2 advantages: 1) it's easier to read (at least for people who are used to `next`), 2) it does not copy the whole (perhaps long) key set into a list.

PS: if you want a list with all dict keys you can also just write `list(self.marker_extrinsics_opt)`
PS2: Be aware that there is no guarantee on what the first element in the set is going to be. Python >= 3.6 (?) dicts are guaranteed to be sorted in _insertion_ order, but we support Python >= 3.4 so we cannot rely on this for now.